File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    count = [0, 0]
    for i in s:
        if count[0] == count[1]:
            answer += 1
            k = i
        
        if k == i:
            count[0] += 1
        else:
            count[1] += 1
    
    return answer

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        
        int answer = 0;
        int[] count = {0, 0};
        char x = s.charAt(0);
        
        for(int i=0; i<s.length(); i++) {
            if (count[0] == count[1]) {
                answer += 1;
                x = s.charAt(i);
            }
            
            if(s.charAt(i) == x) {
                count[0] += 1;
            } else {
                count[1] += 1;
            }
            
        }

        return answer;
    }
}